common/safe_prime.go

```diff
@@ -121,7 +125,7 @@ var smallPrimesProduct = new(big.Int).SetUint64(16294579238595022365)
 // This function generates safe primes of at least 6 `bitLen`. For every
 // generated safe prime, the two most significant bits are always set to `1`
 // - we don't want the generated number to be too small.
-func GetRandomSafePrimesConcurrent(bitLen, numPrimes int, timeout time.Duration, concurrency int) ([]*GermainSafePrime, error) {
+func GetRandomSafePrimesConcurrent(ctx context.Context, bitLen, numPrimes int, concurrency int) ([]*GermainSafePrime, error) {
```

Comment:
This function is public so this is a potential breaking change. Shouldn't we expose ` GetRandomSafePrimesConcurrentWithContext` for full backward compatibility?

Reply:
I was thinking about it but I am not sure if it makes sense to complicate the code. I think `common.GetRandomnSafePrimesConcurrent` and `paillier.GenerageKeyPair` are not something the code using `tss-lib` calls.